src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -0,0 +1,180 @@
+import {SyntheticEventTarget} from '../../../esl-utils/dom/events/target';
+import {ESLMixinElement} from '../../../esl-mixin-element/ui/esl-mixin-element';
+import {overrideEvent} from '../../../esl-utils/dom/events/misc';
+
+type SwipeDirection = 'left' | 'right' | 'up' | 'down';
+type SwipeEventName = 'swipe' | 'swipe:left' | 'swipe:right' | 'swipe:up' | 'swipe:down';
+
+interface SwipeEvent {
+  direction: SwipeDirection;
+  distanceX: number;
+  distanceY: number;
+  touchStartOriginalEvent: PointerEvent;
+  touchEndOriginalEvent: PointerEvent;
+}
+
+interface SwipeEventTargetConfig {
+  threshold: number;
+  units: string;
+  timeout: number;
+}
```

Comment:
Interfaces are usually a part of Public API that the user faces first so we need to make TS/JS Docs for it.